R/locate_credentials.R

```diff
@@ -286,11 +286,27 @@ check_for_user_supplied_profile <- function(profile, file, region, session_token
     return(NULL)
 }
 
+
 check_for_env_vars <- function(region, file, default_region, session_token, verbose) {
 
     if (isTRUE(verbose)) {
       message("Checking for credentials in Environment Variables")
     }
+  
+    identity <- list(arn=Sys.getenv("AWS_ROLE_ARN"),
+                     token_file=Sys.getenv("AWS_WEB_IDENTITY_TOKEN_FILE"))
+    
+    if (!is_blank(identity$arn) && !is_blank(identity$token_file)){
+      creds <- assume_role_with_web_identity(identity$arn, identity$token_file)
+      
+      # this is a tad ugly... (MD)
+      key <- creds$AssumeRoleWithWebIdentityResponse$AssumeRoleWithWebIdentityResult$Credentials$AccessKeyId
+      secret <- creds$AssumeRoleWithWebIdentityResponse$AssumeRoleWithWebIdentityResult$Credentials$SecretAccessKey
+      session_token <- creds$AssumeRoleWithWebIdentityResponse$AssumeRoleWithWebIdentityResult$Credentials$SessionToken
```

Comment:
Should resolve this comment:
Could do it via 
```suggestion
      creds <- creds$AssumeRoleWithWebIdentityResponse$AssumeRoleWithWebIdentityResult$Credentials
      key <- creds$AccessKeyId
      secret <- creds$SecretAccessKey
      session_token <- creds$SessionToken
```

Reply:
Also will need to delete the code comment too